Mathlib/Order/SuccPred/Tree.lean

```diff
@@ -0,0 +1,299 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+
+import Mathlib.Order.SuccPred.Basic
+import Mathlib.Data.Nat.Find
+import Mathlib.Order.Atoms
+import Mathlib.Data.SetLike.Basic
+
+/-!
+# Rooted trees
+
+This file proves basic results about rooted trees, represented using the ancestorship order.
+This is a `PartialOrder`, with `PredOrder` with the immediate parent as a predecessor, and an
+`OrderBot` which is the root. We also have an `IsPredArchimedean` assumption to prevent infinite
+dangling chains.
+
+--/
+
+variable {α : Type*}
+
+variable [PartialOrder α] [PredOrder α]
+
+instance Set.Ici.predOrder [DecidableEq α] {a : α} :
+  PredOrder (Set.Ici a) where
+  pred := fun x ↦ if ha : x.1 = a then ⟨a, by simp⟩ else
+    ⟨Order.pred x.1, Order.le_pred_of_lt <| lt_of_le_of_ne (by simpa using x.2) <| Ne.symm ha⟩
+  pred_le := fun ⟨x, hx⟩ ↦ by dsimp; split <;> simp_all [Order.pred_le]
+  min_of_le_pred := @fun ⟨x, hx⟩ h ↦ by
+    dsimp at h
+    rw [isMin_iff_eq_bot]
+    apply Subtype.val_injective
+    simp only [coe_bot]
+    split at h
+    · assumption
+    · simp only [Subtype.mk_le_mk] at h
+      apply Order.min_of_le_pred at h
+      exact (h.eq_of_le hx).symm
+  le_pred_of_lt := @fun ⟨b, hb⟩ ⟨c, hc⟩ h ↦ by
+    rw [Subtype.mk_lt_mk] at h
+    dsimp only
+    split
+    · simp_all [le_of_lt]
+    · exact Order.le_pred_of_lt h
```

Comment:
Shouldn't this be earlier?

Reply:
What do you mean? Earlier than what?

Reply:
In an earlier file, I mean

Reply:
What would be a good location? `SuccPred/Basic`?

Reply:
If it fits there, that would be great. Else it might warrant a new file

Reply:
I think I'll move it to a separate PR. Currently it's stated using `s.OrdConnected`, but that's not the "correct" requirement - for instance, in an `IsPredArchimedean` order, all sets can be made into a `PredOrder`.